### ISSUE TITLE
Preserve CSS variables in the Presenters

### ISSUE THREAD

**Issue by MMMalik** [UX-and-I/storybook-design-token#108]
## Type of report

Feature request

## Background

Currently, values of CSS variables are extracted at build time and presenters use fixed values. This works fine for compile-time variables such as Sass variables but imposes unnecessary limitations in case of CSS variables.

Example:
<img width="383" alt="image" src="https://user-images.githubusercontent.com/12546548/170719896-570eb186-174c-4d32-bc3b-88f55fe18ffa.png">

I would expect `background: hsl(240,100%,50%);` to be `background: var(--b500);`.

## Provide description of the new feature

An option such as `preserveCSSVariables` can be introduced to the addon, so that [determineTokenValue function](https://github.com/UX-and-I/storybook-design-token/blob/master/addon/src/parsers/postcss.parser.ts#L156) can skip value extraction for CSS variables. Suggested config:

```js
module.exports = {
  ...,
  addons: [
    ...,
    {
      name: 'storybook-design-token',
      options: {
        preserveCSSVariables: true
      }
    }
  ],
};
```

## Provide rationale for this change

I see two possible scenarios where this might be useful:

1. Theme is somehow replaced on runtime, e.g. user click a button to switch theme to dark, etc. Presenters should be able to show up-to-date values.
2. Stylesheet with CSS variables can be loaded separately from tokens. This currently leads to empty Presenters. This is actually the case in one of the projects I'm working on and I can't find a workaround. See minimal repro in the tweaked [demo-webpack5](https://github.com/MMMalik/storybook-design-token/tree/demo-css-vars/demo-webpack5). This is the outcome:
<img width="758" alt="image" src="https://user-images.githubusercontent.com/12546548/170718313-34ebe051-010a-432a-9382-dd1f191abbf0.png">

## Other info

This is a great project! Thanks for your work. I can provide a PR if you're interested.

## Related issues: 

- https://github.com/UX-and-I/storybook-design-token/issues/32
- https://github.com/UX-and-I/storybook-design-token/issues/106

**Comment by prodkt** [comment]
> 1. Theme is somehow replaced on runtime, e.g. user click a button to switch theme to dark, etc. Presenters should be able to show up-to-date values.

🔥🔥🔥 this! We're managing a multi-brand/multi-themed design system. This was the hang-up that stopped us from using this add-on. Had this been in place it would have been a clear decision forward for us.

@MMMalik I see there is a PR open for this potential fix. Are you using this fix in your environment and is it working stable for you? And have you any idea if the developers plan to push this through?

**Comment by MMMalik** [comment]
> I see there is a PR open for this potential fix. Are you using this fix in your environment and is it working stable for you? And have you any idea if the developers plan to push this through?

I've tested the fix within my project and it's working fine. However, I haven't used in production yet.

I've no idea when my PR might be merged though, or even reviewed, as I can see that things have been going rather slowly in this repo recently. Perhaps @Sqrrl could shed some light? I hope it will get some attention from project mainteiners soon since given reactions to this issue it looks like this change is anticipated by many.

**Comment by Sqrrl** [comment]
Hey, thank you for your PR and feedback. Published as v2.5.0.
